src/content/docs/apm/agents/net-agent/net-agent-api/net-agent-api.mdx

```diff
@@ -0,0 +1,2579 @@
+  - /docs/agents/net-agent/net-agent-api/recordresponsetimemetric-net-agent
+  - /docs/agents/net-agent/net-agent-api/record-response-time-metric
+  - /docs/apm/agents/net-agent/net-agent-api/recordresponsetimemetric-net-agent-api
+  - /docs/agents/net-agent/net-agent-api/set-application-name
+  - /docs/agents/net-agent/net-agent-api/setapplicationname-net-agent
+  - /docs/apm/agents/net-agent/net-agent-api/set-application-name
+  - /docs/agents/net-agent/net-agent-api/set-transaction-uri
+  - /docs/agents/net-agent/net-agent-api/settransactionuri
+  - /docs/apm/agents/net-agent/net-agent-api/set-transaction-uri
+  - /docs/agents/net-agent/net-agent-api/set-user-parameters
+  - /docs/agents/net-agent/net-agent-api/setuserparameters-net-agent
+  - /docs/apm/agents/net-agent/net-agent-api/set-user-parameters
+  - /docs/agents/net-agent/net-agent-api/seterrorgroupcallback-net-agent-api
+  - /docs/agents/net-agent/net-agent-api/seterrorgroupcallback-net-agent
+  - /docs/agents/net-agent/net-agent-api/set-error-group-callback
+  - /docs/apm/agents/net-agent/net-agent-api/seterrorgroupcallback-net-agent-api
+  - /docs/agents/net-agent/net-agent-api/setllmtokencountingcallback-net-agent-api
+  - /docs/agents/net-agent/net-agent-api/setllmtokencountingcallback-net-agent
+  - /docs/agents/net-agent/net-agent-api/set-llm-token-counting-callback
+  - /docs/apm/agents/net-agent/net-agent-api/setllmtokencountingcallback-net-agent-api
+  - /docs/agents/net-agent/net-agent-api/settransactionname-net-agent-api
+  - /docs/agents/net-agent/net-agent-api/settransactionname-net-agent
+  - /docs/agents/net-agent/net-agent-api/set-transaction-name
+  - /docs/apm/agents/net-agent/net-agent-api/settransactionname-net-agent-api
+  - /docs/agents/net-agent/net-agent-api/start-agent
+  - /docs/agents/net-agent/net-agent-api/startagent-net-agent
+  - /docs/apm/agents/net-agent/net-agent-api/start-agent
+  - /docs/agents/net-agent/net-agent-api/tracemetadata-net-agent-api-0
+  - /docs/apm/agents/net-agent/net-agent-api/tracemetadata-net-agent-api-0
+
+freshnessValidatedDate: never
+---
+
+New Relic's .NET agent includes an API that allows you to extend the agent's standard functionality. For example, you can use the .NET agent API for:
+
+* Customizing your app name
+* Creating custom transaction parameters
+* Reporting custom errors and metrics
+
+You can also customize some of the .NET agent's default behavior by adjusting [configuration settings](/docs/agents/net-agent/configuration/net-agent-configuration) or using [custom instrumentation](/docs/agents/net-agent/custom-instrumentation/introduction-net-custom-instrumentation).
+
+## Requirements
+
+To use the .NET agent API, make sure you have the [latest .NET agent release](/docs/release-notes/agent-release-notes/net-release-notes). Then, add a reference to the agent in your project using one of the two options below:
+
+   * Add a reference to `NewRelic.Api.Agent.dll` to your project.
+    OR
```

Comment:
I think this needs an extra line break or something to show up on its own line, in preview it shows up on same line

![Uploading image.png…]()